README.md
# https-www.pythonanywhere.com-user-gianpieropalmieri-consoles-9964535-


history.txt
    1  cd mysite
    2  git status
    3  git add .
    4  git commit -m "dans la bible quel est le fleuve dont les eaux ont ete changees en sang"
    5  mkvirtualenv flask-tutorial --python=python3.6
    6  pip install flask flask-login flask-sqlalchemy mysql-connector-python
    7  pip freeze > requirements.txt
    8  git status
    9  git add .
   10  git commit -m "la lune est le planete de l'observatoire du vatican des jesuites"
   11  pip install Flask-Migrate
   12  export FLASK_APP=flask_app.py
   13  flask db inti
   14  flask db init
   15  flask db migrate
   16  flask db stamp head
   17  pip freeze > requirements.txt
   18  git status
   19  git add .
   20  git commit -m "c'est la bonne reponse en plus tout est parfait, madeleine"
   21  flask db migrate
   22  flask db upgrade
   23  git status
   24  git add .
   25  git commit -m "puisque vous etes pilote, decollez"
   26  flask db migrate
   27  flask db upgrade
   28  flask shell
   29  echo "# https-www.pythonanywhere.com-user-gianpieropalmieri-consoles-9964535-" >> README.md
   30  git init
   31  git add README.md
   32  git commit -m "first commit"
   33  git remote add origin https://github.com/kevinradjit/https-www.pythonanywhere.com-user-gianpieropalmieri-consoles-9964535-.git
   34  git push -u origin master
   35  history 
   36  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="gianpieropalmier",
    password="Exeter22",
    hostname="gianpieropalmieri.mysql.pythonanywhere-services.com",
    databasename="gianpieropalmier$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "46546549879781321321321"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/898cb335dfcc_.py
"""empty message

Revision ID: 898cb335dfcc
Revises: 
Create Date: 2018-08-25 22:37:52.431821

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '898cb335dfcc'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/d5fcf3c2e0fb_.py
"""empty message

Revision ID: d5fcf3c2e0fb
Revises: 898cb335dfcc
Create Date: 2018-08-25 22:40:17.998958

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'd5fcf3c2e0fb'
down_revision = '898cb335dfcc'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###
